helpers/fetchApiData.ts

```diff
@@ -0,0 +1,42 @@
+export function validateListPath(path: string) {
+  const regex =
+    /^lists\/[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}(\/(overview|activity|contributors)\?limit=\d+)?$/;
+
+  return regex.test(path);
+}
+
+export async function fetchApiData<T>({
```

Comment:
A lot of this code is similar to the `publicFetcher`, so maybe that makes more sense to just use that. I had started that prior to knowing that existed.

Reply:
We can refactor this after if its already working